docs/data/material/getting-started/templates/blog/Blog.js

```diff
@@ -41,25 +41,26 @@ export default function Blog() {
   };
 
   return (
-    <ThemeProvider theme={showCustomTheme ? blogTheme : defaultTheme}>
-      <CssBaseline />
-      {/* you can delete this NavBar component since it's just no navigate to other pages */}
-      <NavBar
-        toggleCustomTheme={toggleCustomTheme}
-        showCustomTheme={showCustomTheme}
-        mode={mode}
-        toggleColorMode={toggleColorMode}
-      />
-      <AppAppBar />
-      <Container
-        maxWidth="lg"
-        component="main"
-        sx={{ display: 'flex', flexDirection: 'column', mt: 24, mb: 16, gap: 4 }}
-      >
-        <MainContent />
-        <Latest />
-      </Container>
-      <Footer />
-    </ThemeProvider>
+    // You can delete this TemplateFrame component since it's just no navigate to other pages
+    <TemplateFrame
+      toggleCustomTheme={toggleCustomTheme}
+      showCustomTheme={showCustomTheme}
+      mode={mode}
+      toggleColorMode={toggleColorMode}
+    >
+      <ThemeProvider theme={showCustomTheme ? blogTheme : defaultTheme}>
```

Comment:
We already render a `ThemeProvider` in `TemplateFrame`. Could we apply this logic there passing the `theme` as a prop to `<TemplateFrame>`?

Reply:
The issue with that approach is that the `TemplateFrame` would end up using both themes. Since it’s not part of the actual template content, I didn’t want it to change the theme, which could cause layout shifts in the navigation menu.

Reply:
Got it!